addons/website_product_comparison/static/src/js/website_product_comparison.js

```diff
@@ -0,0 +1,342 @@
+    product_data: {},
+    init: function() {},
+    show: function() {
+        $('.o_product_feature_panel').show();
+    },
+    load_products:function(product_ids) {
+        var self = this;
+        return ajax.jsonRpc('/shop/get_product_data', 'call', {
+            'product_ids': product_ids
+        }).then(function (data) {
+            _.each(data, function(product) {
+                self.product_data[product.product.id] = product;
+            });
+        });
+    },
+    animate_clone: function(cart, $elem, offset_top, offset_left) {
+        cart.find('.o_product_circle').addClass('o_red_highlight o_shadow_animation').delay(500).queue(function(){
+            $(this).removeClass("o_shadow_animation").dequeue();
+        });
+        var imgtodrag = $elem.find('img').eq(0);
+        if (imgtodrag.length) {
+            var imgclone = imgtodrag.clone()
+            .offset({
+                top: imgtodrag.offset().top,
+                left: imgtodrag.offset().left
+            })
+            .addClass('o_product_comparison_animate')
+            .appendTo($('body'))
+            .animate({
+                'top': cart.offset().top + offset_top,
+                'left': cart.offset().left + offset_left,
+                'width': 75,
+                'height': 75
+            }, 1000, 'easeInOutExpo');
+
+            imgclone.animate({
+                'width': 0,
+                'height': 0
+            }, function () {
+                $(this).detach();
+            });
+        }
+    }
+});
+
+var ProductComparison = Widget.extend({
+    template:"product_comparison_template",
+    events: {
+        'click .o_product_panel_header': 'toggle_panel',
+    },
+    init: function(parent){
+        this.parent = parent;
+        this.comparelist_product_ids = JSON.parse(utils.get_cookie('comparelist_product_ids') || 'null') || [];
+        this.product_compare_limit = 4;
+        this.parent.show();
+    },
+    start:function(){
+        var self = this;
+        self.popover = $('#comparelist .o_product_panel_header').popover({
+            trigger: 'manual',
+            animation: true,
+            html: true,
+            title: function () {
+                return _t("Compare Products");
+            },
+            container: '.o_product_feature_panel',
+            placement: 'top',
+            template: '<div style="width:600px;" class="popover comparator-popover" role="tooltip"><div class="arrow"></div><h3 class="popover-title"></h3><div class="popover-content"></div></div>',
+            content: function() {
+                return $('#comparelist .o_product_panel_content').html();
+            }
+        });
+        $('.oe_website_sale .o_add_compare, .oe_website_sale .o_add_compare_dyn').click(function (e){
+            self.parent.animate_clone($('#comparelist .o_product_panel_header'), $(this).closest('form'), -50, 10);
+            if (self.comparelist_product_ids.length < self.product_compare_limit) {
```

Comment:
careful, number of products in cookies is not the same as displayed (cf screenshot below)